### PR TITLE
290 user sign up form validation

### PR DESCRIPTION
**Pull request by yulejessica**
## Dev Summary

The user sign-up form should validate accepted terms, email, and zip and provide useful validation errors to the user for each of those fields.
This will require using a form library (I used the UserOrg sign-up form as a reference ) as well as a multi-step verification form (when the user can go to the next step)

## Test Plan

repro steps:

1. localhost:3000 
2. click X button
expected: 

## Resources

- [stackoverflow is cool](stackoverflow.com)

### CODE REVIEW

File: client/src/components/Users/Auth/SignUpCitizen/SignUpCitizen.tsx [hunk 1]
```diff
@@ -0,0 +1,590 @@
...
+      setUser(data);
+      handleNext();
+    }
+  };
+
+  return (
+    <div className="SignupCitizen">
+      {/* <p>dirtyFields: {JSON.stringify(dirtyFields)}</p>
+      <p>isValid: {isValid.toString()}</p> */}
+      <Grid container>
```
Comment (esteban-gs) [inline client/src/components/Users/Auth/SignUpCitizen/SignUpCitizen.tsx:141]:
```suggestion
  return (
    <div className="SignupCitizen">
      <Grid container>
```

Comment (esteban-gs) [inline client/src/components/Users/Auth/SignUpCitizen/SignUpCitizen.tsx:141]:
```suggestion
  return (
    <div className="SignupCitizen">
      <Grid container>
```

File: client/src/components/Users/Auth/SignUpCitizen/validation-rules.ts [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+  vals: FormData,
+): boolean => {
+  const errVals = Object.values(errors);
+  const valsAsStringArr = Object.values(vals);
+  return (
+    errVals.every((x) => x === undefined) &&
+    valsAsStringArr.length === 17 &&
+    valsAsStringArr.every((s) => !!s)
+  );
+};
```
Comment (esteban-gs) [inline client/src/components/Users/Auth/SignUpCitizen/validation-rules.ts:43]: We no longer need this one, right? It's not used anywhere else
```suggestion
```

File: client/src/views/SearchResults.tsx [hunk 3]
```diff
@@ -128,7 +128,7 @@ function SearchResults(): JSX.Element {
 
   React.useEffect(() => {
     fetchSearchData();
-  }, [querySearchText, querySearchCategory]);
+  }, [querySearchText, querySearchCategory, fetchSearchData]);
```
Comment (esteban-gs) [inline client/src/views/SearchResults.tsx:131]: I don't think we need `fetchSearchData` here right?

File: client/src/views/TermsOfService.tsx
Comment (esteban-gs) [inline client/src/views/TermsOfService.tsx:1]: @yulejessica , are the style changes to this component necessary?

**Review by esteban-gs — CHANGES_REQUESTED** [review]

File: client/src/views/TermsOfService.tsx
Comment (yulejessica) [reply client/src/views/TermsOfService.tsx:1]: Probably not, but I added to make the footer not stick to the top.

File: client/src/views/SearchResults.tsx (hunk 3 above)
Comment (esteban-gs) [reply client/src/views/SearchResults.tsx:131]: ok. I think it's ok to leave it as you have it then

**Review by esteban-gs — APPROVED** [review]
Looks good!

**Pull request merged**